File: leetcode/algorithms/remove-duplicates-from-sorted-array.js
const removeDuplicates = (nums) => {
  let lastUniqIdx = 0

  for (let idx = 1; idx < nums.length; idx++) {
    if (nums[lastUniqIdx] !== nums[idx]) {
      lastUniqIdx += 1
      nums[lastUniqIdx] = nums[idx]
    }
  }

  return lastUniqIdx + 1
}

File: leetcode/algorithms/remove-duplicates-from-sorted-array.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        last_uniq_idx = 0
        for idx in range(1, len(nums)):
            if nums[last_uniq_idx] != nums[idx]:
                last_uniq_idx += 1
                nums[last_uniq_idx] = nums[idx]

        return last_uniq_idx + 1
